Java/core/src/main/java/com/amazon/randomcutforest/imputation/ImputeVisitor.java

```diff
@@ -32,26 +32,65 @@
  * value is the anomaly score for the imputed point.
  */
 public class ImputeVisitor implements MultiVisitor<double[]> {
-    private final boolean[] missing;
-    private final boolean[] liftedMissing;
-    private double[] queryPoint;
-    private double[] liftedPoint;
-    private double rank;
+
+    // default large values for initialization; consider -ve log( 0 )
+    public static double DEFAULT_INIT_VALUE = Double.MAX_VALUE / 4;
```

Comment:
Why this choice specifically? Because we need a big number but not the biggest number?

Reply:
Yes. We may make approximate comparisons of rank (say up to factors of 4) and this avoids any potential overflows.